tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNet/AspNetMvcIntegration.cs

```diff
@@ -178,10 +175,10 @@ public static Scope CreateScope(ControllerContextStruct controllerContext)
 
                     tags.SetAnalyticsSampleRate(IntegrationId, tracer.Settings, enabledWithGlobalSetting: true);
 
-                    if (newResourceNamesEnabled)
+                    if (newResourceNamesEnabled && string.IsNullOrEmpty(controllerContext.HttpContext.Items[SharedItems.HttpContextPropagatedResourceNameKey] as string))
```

Comment:
nit: no need to access `HttpContext` again
```suggestion
                    if (newResourceNamesEnabled && string.IsNullOrEmpty(httpContext.Items[SharedItems.HttpContextPropagatedResourceNameKey] as string))
```